### PR TITLE
Implement amp-byside-content extension improvements

### PR DESCRIPTION
**Pull request by bysidedevel3rdparty**
This PR implements improvements to amp-byside-content extension based on code review suggestions discussed on #12798 .

### CODE REVIEW

**Comment by aghassemi** [comment]
@bysidedevel3rdparty Thanks! I guess we both did the same, #13361 handles the unit tests without prod code changes, let's merge that and then please rebase, resolve conflicts and keep the remaining changes in this PR.

**Comment by aghassemi** [comment]
@bysidedevel3rdparty on second thought, maybe just copy the unit test changes and the removal of the whole `win.AMP_TEST ? ` from #13361 into this PR and I will close that one.

**Comment by bysidedevel3rdparty** [comment]
Awesome, we will copy the unit test changes and update this PR right away.

**Comment by bysidedevel3rdparty** [comment]
Unit tests have been combined with previously provided code changes.

It now seems that the travis ci check is somehow stuck and not returning as valid test.

File: extensions/amp-byside-content/0.1/amp-byside-content.js [hunk 1]
```diff
@@ -122,8 +122,8 @@ export class AmpBysideContent extends AMP.BaseElement {
    * @override
    */
   preconnectCallback(onLayout) {
-    if (this.iframeSrc_) {
-      this.preconnect.url(this.iframeSrc_, onLayout);
+    if (this.iframeSrc_ || this.origin_) {
```
Comment (jridgewell) [inline extensions/amp-byside-content/0.1/amp-byside-content.js]: Should be able to compose the origin here, but it'll require updating `#composeOrigin_` a bit to grab the subdomain.

File: extensions/amp-byside-content/0.1/amp-byside-content.js [hunk 2]
```diff
@@ -270,58 +256,66 @@ export class AmpBysideContent extends AMP.BaseElement {
...
    *    requested dimensions.
-   * @param {Object} data
    * @private
    */
-  updateSize_(data) {
-    // Calculate new height of the container to include the padding.
-    // If padding is negative, just use the requested height directly.
+  updateSize_() {
+    const args = Array.prototype.slice.call(arguments);
+    const data = args.length ? args[0] : null;
```
Comment (jridgewell) [inline extensions/amp-byside-content/0.1/amp-byside-content.js]: What is this change for?

File: extensions/amp-byside-content/0.1/amp-byside-content.js [hunk 3]
```diff
@@ -270,58 +256,66 @@ export class AmpBysideContent extends AMP.BaseElement {
...
     if (!data) {
       return;
     }
 
-    let newHeight;
-    const height = parseInt(data['height'], 10);
-    if (!isNaN(height)) {
-      newHeight = Math.max(
-          height + (this.element./*OK*/offsetHeight
+    this.getVsync().mutate(() => {
```
Comment (jridgewell) [inline extensions/amp-byside-content/0.1/amp-byside-content.js]: This needs to be a measure phase, not the mutate phase. `this.getVsync().measure(() => {})`

File: extensions/amp-byside-content/0.1/amp-byside-content.js (hunk 1 above)
Comment (bysidedevel3rdparty) [reply extensions/amp-byside-content/0.1/amp-byside-content.js]: Could not get what you mean here. preconnectCallback is called after build, therefore this.origin_ is always correctly defined.

File: extensions/amp-byside-content/0.1/amp-byside-content.js (hunk 2 above)
Comment (bysidedevel3rdparty) [reply extensions/amp-byside-content/0.1/amp-byside-content.js]: This change was made due to debounce check types returning an error. Reverted and changed how debounce function was called.

**Comment by bysidedevel3rdparty** [comment]
lgtm analysis is failing, returning the message "Analysis Failed (could not build the base commit (433685a))". Can you explain why?

**Comment by jridgewell** [comment]
> lgtm analysis is failing

Not sure why. Can you try rebasing the branch off of master?

**Comment by aghassemi** [comment]
@bysidedevel3rdparty Could you please rebase this PR?

**Comment by aghassemi** [comment]
Closing the PR for house-keeping reasons, please reopen after rebase. Thanks and apologies.

**Pull request closed without merging**